### ISSUE TITLE
Overlapping instance error when building pairing

### ISSUE THREAD

**Issue by jkachmar** [adjoint-io/pairing#10]
`pairing` fails to build via `cabal new-build` with the following error:

<details><summary>Build Log</summary>
<p>

```
src/Pairing/Fr.hs:28:10: error:
    • Overlapping instances for Ord Fr
        arising from a use of ‘ghc-prim-0.5.3:GHC.Classes.$dm<’
      Matching instances:
        instance Ord (PrimeField p) -- Defined in ‘PrimeField’
        instance Ord Fr -- Defined at src/Pairing/Fr.hs:28:10
    • In the expression: ghc-prim-0.5.3:GHC.Classes.$dm< @(Fr)
      In an equation for ‘<’: (<) = ghc-prim-0.5.3:GHC.Classes.$dm< @(Fr)
      In the instance declaration for ‘Ord Fr’
   |
28 | instance Ord Fr where
   |          ^^^^^^

src/Pairing/Fr.hs:28:10: error:
    • Overlapping instances for Ord Fr
        arising from a use of ‘ghc-prim-0.5.3:GHC.Classes.$dm<=’
      Matching instances:
        instance Ord (PrimeField p) -- Defined in ‘PrimeField’
        instance Ord Fr -- Defined at src/Pairing/Fr.hs:28:10
    • In the expression: ghc-prim-0.5.3:GHC.Classes.$dm<= @(Fr)
      In an equation for ‘<=’:
          (<=) = ghc-prim-0.5.3:GHC.Classes.$dm<= @(Fr)
      In the instance declaration for ‘Ord Fr’
   |
28 | instance Ord Fr where
   |          ^^^^^^

src/Pairing/Fr.hs:28:10: error:
    • Overlapping instances for Ord Fr
        arising from a use of ‘ghc-prim-0.5.3:GHC.Classes.$dm>’
      Matching instances:
        instance Ord (PrimeField p) -- Defined in ‘PrimeField’
        instance Ord Fr -- Defined at src/Pairing/Fr.hs:28:10
    • In the expression: ghc-prim-0.5.3:GHC.Classes.$dm> @(Fr)
      In an equation for ‘>’: (>) = ghc-prim-0.5.3:GHC.Classes.$dm> @(Fr)
      In the instance declaration for ‘Ord Fr’
   |
28 | instance Ord Fr where
   |          ^^^^^^

src/Pairing/Fr.hs:28:10: error:
    • Overlapping instances for Ord Fr
        arising from a use of ‘ghc-prim-0.5.3:GHC.Classes.$dm>=’
      Matching instances:
        instance Ord (PrimeField p) -- Defined in ‘PrimeField’
        instance Ord Fr -- Defined at src/Pairing/Fr.hs:28:10
    • In the expression: ghc-prim-0.5.3:GHC.Classes.$dm>= @(Fr)
      In an equation for ‘>=’:
          (>=) = ghc-prim-0.5.3:GHC.Classes.$dm>= @(Fr)
      In the instance declaration for ‘Ord Fr’
   |
28 | instance Ord Fr where
   |          ^^^^^^

src/Pairing/Fr.hs:28:10: error:
    • Overlapping instances for Ord Fr
        arising from a use of ‘ghc-prim-0.5.3:GHC.Classes.$dmmax’
      Matching instances:
        instance Ord (PrimeField p) -- Defined in ‘PrimeField’
        instance Ord Fr -- Defined at src/Pairing/Fr.hs:28:10
    • In the expression: ghc-prim-0.5.3:GHC.Classes.$dmmax @(Fr)
      In an equation for ‘max’:
          max = ghc-prim-0.5.3:GHC.Classes.$dmmax @(Fr)
      In the instance declaration for ‘Ord Fr’
   |
28 | instance Ord Fr where
   |          ^^^^^^

src/Pairing/Fr.hs:28:10: error:
    • Overlapping instances for Ord Fr
        arising from a use of ‘ghc-prim-0.5.3:GHC.Classes.$dmmin’
      Matching instances:
        instance Ord (PrimeField p) -- Defined in ‘PrimeField’
        instance Ord Fr -- Defined at src/Pairing/Fr.hs:28:10
    • In the expression: ghc-prim-0.5.3:GHC.Classes.$dmmin @(Fr)
      In an equation for ‘min’:
          min = ghc-prim-0.5.3:GHC.Classes.$dmmin @(Fr)
      In the instance declaration for ‘Ord Fr’
   |
28 | instance Ord Fr where
   |          ^^^^^^
```

</p>
</details>

**Comment by Multramate** [comment]
The new release `galois-field-0.4.0` has these instances but those in `pairing` haven’t been deleted yet - will fix this in a new version as soon as I can.

**Comment by sumo** [comment]
This is affecting the versions released on Hackage due to the update to galois-field. Upper bound on `galois-field >=0.2 && <0.4.0` resolves the hackage build for pairing 0.4.1

**Comment by sdiehl** [comment]
This patch landed up in Hackage.
http://hackage.haskell.org/package/pairing-0.4.2
